src/classLibrary/Utility/FileFactory.ts

```diff
@@ -0,0 +1,114 @@
+import { isBinaryFile } from 'isbinaryfile';
+import {
+    DirectorySystemInstance,
+    FileContent,
+    FileSystemInstance,
+    GlobSyncOptions,
+    IFileFactory,
+    IFileSystemInstanceMetadata,
+    Ignore,
+    VirtualFileSystemInstance
+} from "../interfaces/interfaces";
+import _glob from "glob";
+
+class FileFactory implements IFileFactory {
+    readonly FromRoot: string;
+    readonly ToRoot: string;
+
+    constructor(fromRoot: string, toRoot: string) {
+        this.FromRoot = fromRoot;
+        this.ToRoot = toRoot;
+    }
+
+    /**
+     * Generates the file instance according to the relative path to the file/directory, directory from the source root path, destination root path
+     * returns the file
+     * @param relativePath relative path from the source root path including the optional filepaths
+     * @param from Optional filepath of the file/directory from the source root filepath
+     * @param to Optional filepath of the file/directory from the destination root filepath
+     * @param ignore Optional ignore for the resolvers to ignore content or filepath
+     */
+    CreateFileSystemInstance(relativePath: string, from: string = "./", to: string = './', ignore?: Ignore): VirtualFileSystemInstance {
+        const absFrom = path.resolve(this.FromRoot, from, relativePath);
+        const absTo = path.resolve(this.ToRoot, to, relativePath);
+        const fileMetadata: IFileSystemInstanceMetadata = {
+            relativePath: relativePath,
+            sourceAbsolutePath: absFrom,
+            destinationAbsolutePath: absTo,
+        };
+        let ignoreConfig: Ignore = {
+            variableResolver: {},
+            inlineResolver: {},
+            ifElseResolver: {},
+            guidResolver: {},
+            custom: {}
+        }
+        if (ignore !== undefined) {
+            ignoreConfig = ignore;
+        }
+        if (fs.lstatSync(absFrom).isFile()) {
+            const file: FileSystemInstance = {
+                metadata: fileMetadata,
+                content: FileContent.String(""),
+                ignore: ignoreConfig
+            };
+            return VirtualFileSystemInstance.File(file);
+        } else {
+            const folder: DirectorySystemInstance = {
+                metadata: fileMetadata,
+                ignore: ignoreConfig
+            }
+            return VirtualFileSystemInstance.Folder(folder);
+        }
+    }
+
+    GetAbsoluteFilePathsOfFileInDestinationPath(fileName: string, fromRoot: string = "./", pattern: string = "**/*.*", ignore?: string | string[]): string[] {
+        let pathPattern = path.resolve(this.ToRoot, fromRoot, pattern);
+		let opts: GlobSyncOptions = {dot: true}
+        if (ignore != null) {
+			opts.ignore = ignore;
+		}
+        //returns all absolute path according to the glob pattern 
+        let files: string[] = _glob.sync(pathPattern, opts);
+        return files.filter((path: string) => path.includes(fileName));
+    }
```

Comment:
What's the purpose of this method? I don't think we need it under IFileFactory.

For npm plug-in we can think about a better way to implement it later, but I don't think this method should be needed.